packages/core/custom-branding/core-custom-branding-server/types.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { KibanaRequest } from '@kbn/core-http-server';
+import { CustomBranding } from '@kbn/core-custom-branding-common';
+
+/** @public */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface CustomBrandingStart {}
+
+export type CustomBrandingFetchFn = (request: KibanaRequest) => Promise<CustomBranding>;
```

Comment:
NIT: it's usually a good practice to not forces external consumers to return promises from hooks, even if the sole implementation does.

```ts
CustomBrandingFetchFn = (request: KibanaRequest) => Promise<CustomBranding>;

=> 

CustomBrandingFetchFn = (request: KibanaRequest) => MaybePromise<CustomBranding>;
```

Reply:
TIL